metals/src/main/scala/scala/meta/internal/metals/BuildTargets.scala

```diff
@@ -469,7 +488,12 @@ final class BuildTargets private (
           val path = resource.toAbsolutePath
           classpaths.collectFirst {
             case (id, classpath) if classpath.contains(path) =>
-              InferredBuildTarget(path, toplevel.value, id)
+              InferredBuildTarget(
+                path,
+                toplevel.value,
+                id,
+                data.fromOptions(_.findSourceJarOf(path, Some(id))),
```

Comment:
```suggestion
                sourceJarFor(id, path)
```
shouldn't we have the fallback here too?

Reply:
Yep, makes sense! Fixed!